src/Umbraco.Infrastructure/PropertyEditors/RichTextEditorPastedImages.cs

```diff
@@ -56,149 +67,301 @@ public sealed class RichTextEditorPastedImages
 
         _tempFolderAbsolutePath = _hostingEnvironment.MapPathContentRoot(Constants.SystemDirectories.TempImageUploads);
 
+        _imageUrlGenerator = StaticServiceProvider.Instance.GetRequiredService<IImageUrlGenerator>();
+        _contentSettings = StaticServiceProvider.Instance.GetRequiredService<IOptions<ContentSettings>>().Value;
+    }
```

Comment:
This ctor should use `this` instead, to call the new constructor 🙌 

So it should look like this: 

```
    public RichTextEditorPastedImages(
        IUmbracoContextAccessor umbracoContextAccessor,
        ILogger<RichTextEditorPastedImages> logger,
        IHostingEnvironment hostingEnvironment,
        IMediaService mediaService,
        IContentTypeBaseServiceProvider contentTypeBaseServiceProvider,
        MediaFileManager mediaFileManager,
        MediaUrlGeneratorCollection mediaUrlGenerators,
        IShortStringHelper shortStringHelper,
        IPublishedUrlProvider publishedUrlProvider)
        : this(
            umbracoContextAccessor,
            logger,
            hostingEnvironment,
            mediaService,
            contentTypeBaseServiceProvider,
            mediaFileManager,
            mediaUrlGenerators,
            shortStringHelper,
            publishedUrlProvider,
            StaticServiceProvider.Instance.GetRequiredService<IImageUrlGenerator>(),
            StaticServiceProvider.Instance.GetRequiredService<IOptions<ContentSettings>>())
    {
    }
```

Reply:
Fixed